packages/playwright-test/src/index.ts

```diff
@@ -116,7 +116,7 @@ export const test = _baseTest.extend<TestFixtures, WorkerFixtures>({
     const browser = await playwright[browserName].connect(connectOptions.wsEndpoint, {
       headers: {
         'x-playwright-browser': browserName,
-        'x-playwright-launch-options': JSON.stringify(launchOptions),
+        'x-playwright-launch-options': JSON.stringify({ ...launchOptions, headless, channel }),
```

Comment:
We should not override user values with undefined here. Take a look at [this code](https://github.com/microsoft/playwright/blob/main/packages/playwright-test/src/index.ts#L97-L100).